tsaqafah.csl

```diff
@@ -0,0 +1,501 @@
+    </group>
+  </macro>
+  <citation>
+    <layout delimiter="; " suffix=".">
+      <choose>
+        <if match="any" position="ibid-with-locator">
+          <group delimiter=", ">
+            <text term="ibid" font-style="italic"/>
+            <text macro="locator-subsequent"/>
+          </group>
+        </if>
+        <else-if match="any" position="ibid">
+          <text term="ibid" font-style="italic" suffix="."/>
+        </else-if>
+        <else-if match="any" position="subsequent">
+          <group delimiter=", ">
+            <text macro="Author-Short"/>
+            <text macro="title-short"/>
+            <choose>
+              <if match="any" variable="locator">
+                <text macro="locator-subsequent"/>
+              </if>
+            </choose>
+          </group>
+        </else-if>
+        <else>
+          <group delimiter=", ">
+            <text macro="Author"/>
+            <text macro="title"/>
+            <text macro="edition"/>
+          </group>
+          <group delimiter=", " prefix=" (" suffix=")">
+            <choose>
+              <if type="manuscript" match="none">
+                <group>
+                  <choose>
+                    <if type="broadcast personal_communication" match="none">
+                      <text macro="publisher"/>
+                    </if>
+                  </choose>
+                </group>
+              </if>
+            </choose>
+            <text macro="issued"/>
+          </group>
+          <group prefix=", ">
+            <choose>
+              <if match="none" variable="locator">
+                <text macro="journal-page"/>
+              </if>
+              <else>
+                <group delimiter=" ">
+                  <text macro="locator"/>
+                </group>
+              </else>
+            </choose>
+          </group>
+          <text macro="url" prefix=", "/>
+        </else>
+      </choose>
+    </layout>
+  </citation>
+  <bibliography subsequent-author-substitute="____" hanging-indent="true">
+    <sort>
+      <key macro="Author-sort"/>
+      <key variable="issued"/>
+      <key macro="title"/>
+    </sort>
+    <layout suffix=".">
+      <group delimiter=". ">
+        <text macro="Author-sort"/>
+        <text macro="edition"/>
+        <choose>
+          <if type="manuscript" match="none">
+            <group delimiter=", ">
+              <text macro="issued"/>
+              <text macro="title"/>
+              <choose>
+                <if type="broadcast personal_communication interview" match="none">
+                  <text macro="publisher"/>
+                </if>
+              </choose>
+            </group>
+          </if>
+        </choose>
+        <text macro="journal-page"/>
+        <choose>
+          <if match="none" variable="DOI">
+            <text macro="url" prefix=" "/>
+          </if>
+        </choose>
+      </group>
+      <text variable="DOI" prefix=" [https://doi.org/" suffix="]"/>
```

Comment:
@daredmayo, I changed the DOI prefix from `http://dx.doi.org/` to https://doi.org/`, which is nowadays the preferred format.